### PR TITLE
Improvements to reviews

### LINKED ISSUE #223: Allow rating-less reviews
From Em:
"
Just to follow up my request for it to not be a requirement to have minimum one star on the ratings when inputting a review to a show page, as some of the earlier reviews from The Student, of which we have had many, have not listed a star rating. It looks bad if I have to put 1 star"

I for now have asked them to put 1000 as the rating so we can filter them out easily later.

The text also states review has a max rating of 5, which isn't actually enforced, and in my opinion should not be enforced.

### PR DESCRIPTION
**Pull request by mickzijdel**
- Fix star ratings to work with ratings that are not wholes or halves. Fixes #223.
- Add dedicated controller and views to view, create and edit reviews directly
- Add title and original url to reviews.
- Add reviews to all kinds of events and not just shows.

### CODE REVIEW

File: app/helpers/star_helper.rb [hunk 1]
```diff
@@ -15,6 +19,8 @@ def star_rating(rating)
       stars << '<i class="fas fa-star-half-alt" aria-hidden=”true”></i>'
     end
 
+    stars << " (#{rating})" if (rating_decimal != 0.0 && rating_decimal != 0.5)
```
Comment (TaoBi22) [inline app/helpers/star_helper.rb:22]: As discussed, comment here would be helpful

```suggestion
    # Only multiples of 0.5 can be shown in visual stars, so append the numeric rating if the rating is not a multiple of 0.5 (number of stars shown will be rounded down to nearest multiple)
    stars << " (#{rating})" if (rating_decimal != 0.0 && rating_decimal != 0.5)
```

**Review by TaoBi22 — APPROVED** [review]
Couple of nits but otherwise LGTM _(as someone with little working knowledge of Ruby on Rails)_

File: app/controllers/admin/generic_events_controller.rb [hunk 2]
```diff
@@ -29,7 +29,8 @@ def permitted_params
       pictures_attributes: [:id, :_destroy, :description, :image, :access_level, picture_tag_ids: []],
       team_members_attributes: [:id, :_destroy, :position, :user, :user_id, :proposal],
       attachments_attributes: [:id, :_destroy, :name, :file, :access_level, attachment_tag_ids: []],
-      video_links_attributes: [:id, :_destroy, :name, :link, :access_level, :order]
+      video_links_attributes: [:id, :_destroy, :name, :link, :access_level, :order],
+      reviews_attributes: [:id, :_destroy, :title, :url, :body, :rating, :review_date, :organisation, :reviewer, :event_id],
```
Comment (TaoBi22) [inline app/controllers/admin/generic_events_controller.rb:33]: Tiny edge case so feel free to ignore but is it worth being able to note if the review was published in print but not online so no URL? Esp. if anyone wants to add archival reviews - guessing currently URL can just be left blank but might be nice to explicitly differentiate between no URL found (yet) and no URL exists

**Comment by mickzijdel** [comment]
I have fixed the three things suggested by @TaoBi22, but these will be included in a commit in the next pull request.

**Pull request merged**